qiskit/algorithms/gradients/base_sampler_gradient.py

```diff
@@ -0,0 +1,143 @@
+        self._sampler: BaseSampler = sampler
+        self._default_run_options = run_options
+
+    def run(
+        self,
+        circuits: Sequence[QuantumCircuit],
+        parameter_values: Sequence[Sequence[float]],
+        parameters: Sequence[Sequence[Parameter] | None] | None = None,
+        **run_options,
+    ) -> PrimitiveJob:
+        """Run the job of the sampler gradient on the given circuits.
+
+        Args:
+            circuits: The list of quantum circuits to compute the gradients.
+            parameter_values: The list of parameter values to be bound to the circuit.
+            parameters: The Sequence of Sequence of Parameters to calculate only the gradients of
+                the specified parameters. Each Sequence of Parameters corresponds to a circuit in
+                ``circuits``. Defaults to None, which means that the gradients of all parameters in
+                each circuit are calculated.
+            run_options: Backend runtime options used for circuit execution. The order of priority is:
+                run_options in ``run`` method > gradient's default run_options > primitive's default
+                setting. Higher priority setting overrides lower priority setting.
+
+        Returns:
+            Primitive job contains the gradients of the sampling probability. The i-th result
+            corresponds to ``circuits[i]`` evaluated with parameters bound as ``parameter_values[i]``.
+            The j-th quasi-probability distribution in the i-th result corresponds to the gradients of
+            the sampling probability for the j-th parameter in ``circuits[i]``.
+
+        Raises:
+            ValueError: Invalid arguments are given.
+        """
+        # if ``parameters`` is none, all parameters in each circuit are differentiated.
+        if parameters is None:
+            parameters = [None for _ in range(len(circuits))]
+        # Validate the arguments.
+        self._validate_arguments(circuits, parameter_values, parameters)
+        # The priority of run option is as follows:
+        # run_options in `run` method > gradient's default run_options > primitive's default run_options.
+        run_opts = copy(self._default_run_options)
+        run_opts.update(run_options)
+        job = PrimitiveJob(self._run, circuits, parameter_values, parameters, **run_opts)
+        job.submit()
+        return job
+
+    @abstractmethod
+    def _run(
+        self,
+        circuits: Sequence[QuantumCircuit],
+        parameter_values: Sequence[Sequence[float]],
+        parameters: Sequence[Sequence[Parameter] | None],
+        **run_options,
+    ) -> SamplerGradientResult:
+        """Compute the sampler gradients on the given circuits."""
+        raise NotImplementedError()
+
+    def _validate_arguments(
+        self,
+        circuits: Sequence[QuantumCircuit],
+        parameter_values: Sequence[Sequence[float]],
+        parameters: Sequence[Sequence[Parameter] | None] | None = None,
+    ):
```

Comment:
```suggestion
    ) -> None:
```

Reply:
There might be more instances of this further down in the code, please double-check.